### PR TITLE
[admin]: extract runtime and listeners handlers to separate files

### PR DESCRIPTION
**Pull request by rulex123**
Description: extract runtime and listeners handlers from `admin.h|cc` and into separate classes (part of #5505 ) 
Risk Level: low
Testing: pre-existing tests
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: source/server/http/admin.cc [hunk 1]
```diff
@@ -1023,7 +878,7 @@ AdminImpl::AdminImpl(const std::string& profile_path, Server::Instance& server)
            true},
           {"/reset_counters", "reset all counters to zero", StatsHandler::handlerResetCounters,
            false, true},
-          {"/drain_listeners", "drain listeners", MAKE_ADMIN_HANDLER(handlerDrainListeners), false,
+          {"/drain_listeners", "drain listeners", ListenersHandler::handlerDrainListeners, false,
```
Comment (jmarantz) [inline source/server/http/admin.cc:881]: I missed this in the previous refactor, but I think this transformation removes context from the listeners, and you have to instead pass a bunch more args.

This makes the API thicker and more annoying to change.

I would like to propose a variation on this, to do before this PR, where you define a base class AdminContextBase or similar, which carries with it the server_ and potentially other context that may be needed in the future. Make ListenersHandler an implementation of AdminContextBase, inheriting context like `server_` and whatever else may be needed later on.

Then you can also remove this if-statement from place where handlers are called:
```
      if (handler.requires_server_) {
        code = handler.handler_with_server_(path_and_query, response_headers, response,
                                            admin_stream, server_);
      } else {
        code = handler.handler_(path_and_query, response_headers, response, admin_stream);
      }
```
and treat the cases in common.

Comment (rulex123) [reply source/server/http/admin.cc:881]: @jmarantz I understand your point about potentially cluttering the API, however I live the fact that with the current solution the handlers are static and don't hold any state. How about a variation on your variation, as follows:
`static Http::Code handlerRuntime(absl::string_view path_and_query,
                                   Http::ResponseHeaderMap& response_headers,
                                   Buffer::Instance& response, AdminStream&,
                                   AdminContextBase* context = NULL);`

Comment (jmarantz) [reply source/server/http/admin.cc:881]: I don't understand why what you suggest is cleaner, or why you'd need a default-arg considering the handlers are all called from exactly one place.

Comment (rulex123) [reply source/server/http/admin.cc:881]: Sorry, I might have explained myself poorly, let me try again. Basically my thinking was: could we have a class that contains any context that might be needed by the handlers (e.g. `server_`, `profile_path_`), and update the original function for a handler’s callback in such a way that this context is optional?
https://github.com/envoyproxy/envoy/blob/74c8054e84b37e8c315835cc5c2cf39a80631f26/include/envoy/server/admin.h#L89-L91
  
As far as I can see there are handlers that are being added from extensions e.g. https://github.com/envoyproxy/envoy/blob/74c8054e84b37e8c315835cc5c2cf39a80631f26/source/extensions/common/tap/admin.cc#L29-L34 and I don’t think those would care about the context?

The point about things being cleaner referred to having a context class (that contains all state needed by the handlers) vs having to pass multiple args, as you suggested: so we are in agreement on that.

Comment (jmarantz) [reply source/server/http/admin.cc:881]: Seems messier and less powerful than what I suggest. Why do you want to go that way?

Comment (rulex123) [reply source/server/http/admin.cc:881]: @jmarantz Follow-up PR: https://github.com/envoyproxy/envoy/pull/11029 Could you please take a look when you get a chance?
